DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs

```diff
@@ -0,0 +1,269 @@
+            }
+        }
+
+        public int FolderMappingID
+        {
+            get
+            {
+                if (this._folderMappingID == Null.NullInteger)
+                {
+                    if (!string.IsNullOrEmpty(this.Request.QueryString["ItemID"]))
+                    {
+                        int.TryParse(this.Request.QueryString["ItemID"], out this._folderMappingID);
+                    }
+                }
+
+                return this._folderMappingID;
+            }
+        }
+
+        protected override void OnInit(EventArgs e)
+        {
+            base.OnInit(e);
+
+            if (!this.UserInfo.IsSuperUser && !this.UserInfo.IsInRole(this.PortalSettings.AdministratorRoleName))
+            {
+                this.Response.Redirect(Globals.AccessDeniedURL(), true);
+            }
+
+            this.UpdateButton.Text = (this.FolderMappingID == Null.NullInteger) ? Localization.GetString("Add") : Localization.GetString("Update", this.LocalResourceFile);
+            this.CancelHyperLink.NavigateUrl = this._navigationManager.NavigateURL();
+
+            var controlTitle = Localization.GetString("ControlTitle", this.LocalResourceFile);
+            var controlTitlePrefix = (this.FolderMappingID == Null.NullInteger) ? Localization.GetString("New") : Localization.GetString("Edit");
+
+            this.SyncWarningPlaceHolder.Visible = this.FolderMappingID != Null.NullInteger;
+
+            this.ModuleConfiguration.ModuleControl.ControlTitle = string.Format(controlTitle, controlTitlePrefix);
+        }
+
+        protected override void OnLoad(EventArgs e)
+        {
+            base.OnLoad(e);
+
+            this.UpdateButton.Click += this.cmdUpdate_Click;
+
+            try
+            {
+                this.BindFolderMappingSettings();
+
+                if (!this.IsPostBack)
+                {
+                    this.BindFolderProviders();
+
+                    if (this.FolderMappingID != Null.NullInteger)
+                    {
+                        this.BindFolderMapping();
+
+                        if (this.ProviderSettingsPlaceHolder.Controls.Count > 0 && this.ProviderSettingsPlaceHolder.Controls[0] is FolderMappingSettingsControlBase)
+                        {
+                            var folderMapping = this._folderMappingController.GetFolderMapping(this.FolderMappingID);
+                            var settingsControl = (FolderMappingSettingsControlBase)this.ProviderSettingsPlaceHolder.Controls[0];
+                            settingsControl.LoadSettings(folderMapping.FolderMappingSettings);
+                        }
+                    }
+                }
+            }
+            catch (Exception exc)
+            {
+                DotNetNuke.Services.Exceptions.Exceptions.ProcessModuleLoadException(this, exc);
+            }
+        }
+
+        protected void cboFolderProviders_SelectedIndexChanged(object sender, EventArgs e)
```

Comment:
There are many of these small issue, I think I will leave this PR as is, let's merge it and then I will add the stylecop config and do a separate PR to fix here and other files of this branch... It will be a smaller review that way I think...